lib/tasks/validate_organisation_user_permission_migration.rake

```diff
@@ -0,0 +1,8 @@
+namespace :validate_organisation_user_permission do
+  task run_validation: :environment do
+    User.all.each do |user|
+      raise "User #{user.id} has discrepancies with provider relationships" unless user.providers.ids.sort.uniq == user.providers_via_user_permissions.ids.sort.uniq
```

Comment:
Could do `user.providers.ids - user.providers_via_user_permissions.ids == []` to skip the pair of sort operations executed per user.